File: Backend/DataAccess/BillFlow.Models/Entities/Payment.cs
using BillFlow.Models.Shared.Enums;

namespace BillFlow.Models.Entities;

public class Payment
{
    public Guid Id { get; set; }

    public Guid OwnerId { get; set; }

    public User Owner { get; set; } = null!;

    public Guid InvoiceId { get; set; }

    public Invoice Invoice { get; set; } = null!;

    public decimal Amount { get; set; }

    public PaymentMethod Method { get; set; }

    public PaymentStatus Status { get; set; } = PaymentStatus.Completed;

    public DateTime PaymentDate { get; set; }

    public string? Reference { get; set; }

    public string? Notes { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? UpdatedAt { get; set; }
}

File: Frontend/src/domain/billing/payment.ts
export const PaymentStatus = {
  Completed: 1,
  Refunded: 2,
  Cancelled: 3,
} as const;

export type PaymentStatus = (typeof PaymentStatus)[keyof typeof PaymentStatus];

export const PaymentMethod = {
  Cash: 1,
  BankTransfer: 2,
  CreditCard: 3,
  PayPal: 4,
  Stripe: 5,
} as const;

export type PaymentMethod = (typeof PaymentMethod)[keyof typeof PaymentMethod];

export type PaymentRecord = {
  id: string;
  invoiceId: string;
  invoiceNumber: string;
  amount: number;
  method: PaymentMethod;
  status: PaymentStatus;
  paymentDate: string;
  reference?: string | null;
  notes?: string | null;
  createdAt: string;
  updatedAt?: string | null;
};

export { paymentMethodLabel, paymentStatusLabel } from "./payment-labels";
